build.rs

```diff
@@ -1,40 +1,54 @@
-#[cfg(feature = "inline_asm")]
-fn main() {}
-
-#[cfg(all(not(feature = "inline_asm"), not(feature = "stable")))]
 fn main() {
-    compile_error!("Neither feature \"stable\" nor \"nightly\" was set!");
-}
+    println!("cargo:rerun-if-changed=build.rs");
 
-#[cfg(all(not(feature = "inline_asm"), feature = "stable"))]
-fn main() {
-    use std::ffi::OsString;
-    use std::fs;
+    #[cfg(all(feature = "external_asm", windows))]
+    compile_error!("\"external_asm\" feature is not available on windows toolchain!");
 
-    println!("cargo:rerun-if-changed=build.rs");
+    #[cfg(feature = "instructions")]
+    if std::env::var("CARGO_CFG_TARGET_ARCH").unwrap() != "x86_64" {
+        panic!("\"instructions\" feature is only available for x86_64 targets!");
+    }
```

Comment:
What's the reason for the change in https://github.com/rust-osdev/x86_64/pull/179/commits/22356fc6f6ef6ac414e6cdfd27eddc105041bd4f? Why does a compile-time error not work?

Reply:
In the build script every `cfg(target_value)` is for the running architecture, not the target one. Whereas env `CARGO_CFG_TARGET_VALUE` is for what is going to be compiled. They differ when cross-compiling is involved, and it can get confusing, but the bottom line is this:

`instructions` feature can be compiled on any architecture, for only x86_64 targets.
`external_asm` in theory should compile for windows target, as long as we are compiling on linux.

Reply:
Thanks a lot for clarifying, makes perfect sense!